toolkit/pythonit_toolkit/api/service_client.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+import httpx
+from pythonit_toolkit.headers import SERVICE_JWT_HEADER
+from pythonit_toolkit.pastaporto.tokens import generate_token
+
+
+@dataclass
+class ServiceResponse:
+    errors: Optional[Dict[str, Any]]
+    data: Optional[Dict[str, Any]]
+
+
+class ServiceClient:
+    def __init__(
+        self,
+        url: str,
+        issuer: str,
+        audience: str,
+        jwt_secret: str,
+    ):
+        self.url = url
+        self.issuer = issuer
+        self.audience = audience
+        self.jwt_secret = jwt_secret
+
+    async def execute(
+        self,
+        document: str,
+        variables: Optional[Dict[str, Any]] = None,
+    ):
+        token = generate_token(
+            self.jwt_secret, issuer=self.issuer, audience=self.audience
+        )
+
+        async with httpx.AsyncClient() as client:
+            response = await client.post(
+                self.url,
+                data={"query": document, "variables": variables},
+                headers={
+                    SERVICE_JWT_HEADER: token,
+                },
+            )
+
+            data = await response.json()
+            return ServiceResponse(errors=data.get("errors"), data=data.get("data"))
```

Comment:
what happens if the server response is a 400?

I would add a `response.raise_for_status()` so at least we throw an exception if the response isn't a 200

In the future I imagine here we will have automatic retry, or implement a service mesh on AWS :P